src/ext/serial.cpp

```diff
@@ -57,6 +79,23 @@ template<typename T> class ThingSetSerial: public ThingSetStream
         T& ser;
 };
 
+#elif defined(__ZEPHYR__)
+class ThingSetSerial: public ThingSetStream
+{
+    public:
+        ThingSetSerial(const int c): ThingSetStream(c){}
+
+        void enable() {
+            console_init(); 
+        }
+
+    private:
+        bool readable()
+        {
+            return true;
```

Comment:
This should not just return true, but check if there are characters in the buffer. Otherwise it might never leave the process_input() function anymore.